src/token/tests.rs

```diff
@@ -171,44 +170,134 @@ fn test_get_headers_invalid() {
+    let key2 = example_key_two();
+    let invalid_key1 = FakeKey::try_from(vec![0, 0, 0, 0, 0, 0, 0]).expect("invalid test key");
+    let invalid_key2 =
+        FakeKey::try_from(vec![0, 0, 0, 0, 0, 0xDC, 0xAF]).expect("invalid test key");
+    let rng = FakeRng;
+    let aad = example_aad();
+    // Using example_claims doesn't make sense, since they contain a cnf for the key,
+    // but we don't know the CEK at this point.
+    let claims = ClaimsSetBuilder::new()
+        .audience(AUDIENCE.to_string())
+        .build();
+    let encrypted = encrypt_access_token_multiple::<FakeCrypto, FakeRng>(
+        vec![&key1, &key2],
+        claims.clone(),
+        Some(&aad),
+        Some(unprotected_header.clone()),
+        Some(protected_header.clone()),
+        rng,
+    )?;
+    let (unprotected, protected) = get_token_headers(&encrypted).expect("invalid headers");
+    assert_header_is_part_of(&unprotected_header, &unprotected);
+    assert_header_is_part_of(&protected_header, &protected.header);
+    for key in vec![key1, key2] {
+        assert_eq!(
+            &decrypt_access_token_multiple::<FakeCrypto, FakeCrypto>(&key, &encrypted, Some(&aad))?,
+            &claims
+        );
+    }
+    let failed = decrypt_access_token_multiple::<FakeCrypto, FakeCrypto>(
+        &invalid_key1,
+        &encrypted,
+        Some(&aad),
+    );
+    assert!(failed
+        .err()
+        .filter(|x| matches!(x, AccessTokenError::NoMatchingRecipient))
+        .is_some());
+    let failed = decrypt_access_token_multiple::<FakeCrypto, FakeCrypto>(
+        &invalid_key2,
+        &encrypted,
+        Some(&aad),
+    );
+    assert!(failed
+        .err()
+        .filter(|x| matches!(
+            x,
+            AccessTokenError::CoseCipherError(CoseCipherError::DecryptionFailure)
+        ))
+        .is_some());
+    Ok(())
+}
+
+#[test]
+fn test_encrypt_decrypt_match_multiple(
+) -> Result<(), AccessTokenError<<FakeCrypto as CoseEncryptCipher>::Error>> {
+    let (unprotected_header, protected_header) = example_headers();
+    let key1 = example_key_one();
+    let rng = FakeRng;
+    let aad = example_aad();
+    let claims = ClaimsSetBuilder::new().build();
+    let encrypted = encrypt_access_token_multiple::<FakeCrypto, FakeRng>(
+        vec![&key1, &key1],
+        claims,
+        Some(&aad),
+        Some(unprotected_header.clone()),
+        Some(protected_header.clone()),
+        rng,
+    )?;
+    let (unprotected, protected) = get_token_headers(&encrypted).expect("invalid headers");
+    assert_header_is_part_of(&unprotected_header, &unprotected);
+    assert_header_is_part_of(&protected_header, &protected.header);
+    // In the future, this should only be an error in "strict mode".
+    assert!(
+        decrypt_access_token_multiple::<FakeCrypto, FakeCrypto>(&key1, &encrypted, Some(&aad))
+            .err()
+            .filter(|x| matches!(x, AccessTokenError::MultipleMatchingRecipients))
+            .is_some()
+    );
```

Comment:
To make finding testing issues easier, it would probably be better if there was a separate test case for each possible error instead of just making a single test case for both the successful case as well as all error cases.

However, you may ignore this comment if not doing so would cause a large amount of additional work.